tests/api/reports-customers.php

```diff
@@ -0,0 +1,155 @@
+		$this->assertArrayHasKey( 'date_last_active', $schema );
+		$this->assertArrayHasKey( 'date_last_active_gmt', $schema );
+		$this->assertArrayHasKey( 'orders_count', $schema );
+		$this->assertArrayHasKey( 'total_spend', $schema );
+		$this->assertArrayHasKey( 'avg_order_value', $schema );
+	}
+
+	/**
+	 * Test reports schema.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_reports_schema() {
+		wp_set_current_user( $this->user );
+
+		$request    = new WP_REST_Request( 'OPTIONS', $this->endpoint );
+		$response   = $this->server->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['schema']['properties'];
+
+		$this->assertCount( 14, $properties );
+		$this->assert_report_item_schema( $properties );
+	}
+
+	/**
+	 * Test getting reports without valid permissions.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_get_reports_without_permission() {
+		wp_set_current_user( 0 );
+		$response = $this->server->dispatch( new WP_REST_Request( 'GET', $this->endpoint ) );
+		$this->assertEquals( 401, $response->get_status() );
+	}
+
+	/**
+	 * Test getting reports.
+	 *
+	 * @since 3.5.0
+	 */
+	public function test_get_reports() {
+		wp_set_current_user( $this->user );
+		WC_Helper_Reports::reset_stats_dbs();
+
+		$test_customers = array();
+
+		// Create 10 test customers.
+		for ( $i = 1; $i <= 10; $i++ ) {
+			$test_customers[] = WC_Helper_Customer::create_customer( "customer{$i}", 'password', "customer{$i}@example.com" );
+		}
+
+		// Initialize the report lookup table.
+		delete_transient( 'wc_update_350_all_customers' );
+		WC_Admin_Api_Init::customer_lookup_store_init();
+
+		// Create a test product for use in an order.
+		$product = new WC_Product_Simple();
+		$product->set_name( 'Test Product' );
+		$product->set_regular_price( 25 );
+		$product->save();
+
+		// Place an order for the first test customer.
+		$order = WC_Helper_Order::create_order( $test_customers[0]->get_id(), $product );
+		$order->set_status( 'processing' );
+		$order->set_total( 100 );
+		$order->save();
+
+		$request  = new WP_REST_Request( 'GET', $this->endpoint );
+		$request->set_query_params(
+			array(
+				'per_page' => 5,
+				'order'    => 'asc',
+				'orderby'  => 'username',
+			)
+		);
+
+		$response = $this->server->dispatch( $request );
+		$reports  = $response->get_data();
+		$headers  = $response->get_headers();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 5, count( $reports ) );
+		$this->assertArrayHasKey( 'X-WP-Total', $headers );
+		$this->assertEquals( 10, $headers['X-WP-Total'] );
+		$this->assertArrayHasKey( 'X-WP-TotalPages', $headers );
+		$this->assertEquals( 2, $headers['X-WP-TotalPages'] );
+		$this->assertEquals( $test_customers[0]->get_id(), $reports[0]['user_id'] );
+		$this->assertEquals( 1, $reports[0]['orders_count'] );
+		$this->assertEquals( 100, $reports[0]['total_spend'] );
+		$this->assert_report_item_schema( $reports[0] );
```

Comment:
Might be nice to have a few more test cases here around some of the REST API query options like a date based one, maybe one on the name filter too, just because that is quite unique to this endpoint.

Reply:
Great thinking, this teased out a bug in the `name` parameter! Fixed bug in 72a7cea, added tests in c07474e.

Reply:
💯